Objects/abstract.c

```diff
@@ -129,8 +129,9 @@ PyObject_LengthHint(PyObject *o, Py_ssize_t defaultvalue)
         return defaultvalue;
     }
     if (!PyLong_Check(result)) {
-        PyErr_Format(PyExc_TypeError, "__length_hint__ must be an integer, not %.100s",
-            Py_TYPE(result)->tp_name);
+        PyErr_Format(PyExc_TypeError,
+                     "%T.__length_hint__ must return type int (not %T)",
```

Comment:
```suggestion
                     "%T.__length_hint__() must return type int (not %T)",
```
ditto for other similar cases.